File: Source/Extensions/Tessa.Extensions.Default.Client/UI/KrProcess/StageHandlers/TypedTaskUIHandler.cs
#nullable enable

using System;
using System.Threading.Tasks;
using Tessa.Extensions.Default.Shared.Workflow.KrProcess;
using Tessa.Platform.Storage;
using Tessa.Platform.Validation;
using static Tessa.Extensions.Default.Shared.Workflow.KrProcess.KrConstants;

namespace Tessa.Extensions.Default.Client.UI.KrProcess.StageHandlers
{
    /// <summary>
    /// UI обработчик типа этапа <see cref="StageTypeDescriptors.TypedTaskDescriptor"/>.
    /// </summary>
    public sealed class TypedTaskUIHandler :
        StageTypeUIHandlerBase
    {
        #region Base Overrides

        /// <inheritdoc />
        public override Task Validate(IKrStageTypeUIHandlerContext context)
        {
            if (!context.Row.TryGet<Guid?>(KrTypedTaskSettingsVirtual.TaskTypeID).HasValue)
            {
                context.ValidationResult.AddError(this, "$KrStages_TypedTask_TaskType");
            }

            return Task.CompletedTask;
        }

        #endregion
    }
}

File: WebClient SDK/src/default/ui/krProcess/stageHandlers/typedTaskUIHandler.ts
import {
  IKrStageTypeUIHandlerContext,
  KrStageTypeUIHandler,
  StageTypeHandlerDescriptor,
  typedTaskDescriptor
} from 'tessa/workflow/krProcess';
import { ValidationResult, ValidationResultType } from 'tessa/platform/validation';

import { plainColumnName } from 'tessa/workflow';

/**
 * UI обработчик типа этапа {@link typedTaskDescriptor}.
 */
export class TypedTaskUIHandler extends KrStageTypeUIHandler {
  //#region base overrides

  public descriptors(): StageTypeHandlerDescriptor[] {
    return [typedTaskDescriptor];
  }

  public async validate(context: IKrStageTypeUIHandlerContext): Promise<void> {
    if (!context.row.tryGet(plainColumnName('KrTypedTaskSettingsVirtual', 'TaskTypeID'))) {
      context.validationResult.add(
        ValidationResult.fromText('$KrStages_TypedTask_TaskType', ValidationResultType.Error)
      );
    }
  }

  //#endregion
}
